traffic_portal/app/src/common/modules/dialog/select/DialogSelectController.js

```diff
@@ -23,10 +23,10 @@ var DialogSelectController = function(params, collection, $scope, $uibModalInsta
 
 	$scope.collection = collection;
 
-	$scope.selectedItemId = null;
+	$scope.selectedItemKeyValue = null;
 
 	$scope.select = function() {
-		var selectedItem = _.find(collection, function(item){ return parseInt(item.id) == parseInt($scope.selectedItemId) });
+		var selectedItem = _.find(collection, function(item){ return item[$scope.key] === $scope.selectedItemKeyValue });
```

Comment:
Instead of `_.find`, consider using [the built-in `Array.prototype.find` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) like so:
```javascript
const selectedItem = _.find(collection, function(item){return item[$scope.key] === $selectedItemKeyValue});
```